goalie/adjoint.py

```diff
@@ -534,9 +534,10 @@ def wrapped_solver(subinterval, initial_condition_map, **kwargs):
 
                 # Compute the gradient on the first subinterval
                 if i == 0 and compute_gradient:
-                    self._gradient = controls.delist(
-                        [control.get_derivative() for control in controls]
-                    )
+                    self._gradient = {
+                        field: control.get_derivative()
+                        for field, control in zip(self.fields, controls)
+                    }
```

Comment:
I decided it'd be more useful to have the gradient components in a dictionary, for consistency with how we handle things elsewhere in Goalie.